flexget/components/notify/notifiers/mqtt.py

```diff
@@ -0,0 +1,185 @@
+import json
+
+from loguru import logger
+
+from hashlib import sha256
+import ssl
+
+from flexget import plugin
+#from flexget.config_schema import one_or_more
+from flexget.event import event
+from flexget.plugin import DependencyError, PluginWarning
+
+plugin_name = 'mqtt'
+logger = logger.bind(name=plugin_name)
+
+class MQTTNotifier:
+    """
+    Example::
+      notify:
+        entries:
+          via:
+            - mqtt:
+                broker_address: "iot.eclipse.org"
+                topic: "flexget/notifications"
+                [broker_port: 1883]
+                [broker_timeout: 30]
+                [broker_transport: ['tcp','websockets'] ]
+                [broker_protocol: ['MQTTv31', 'MQTTv311'] ]
+                [username: yourUsernameHere]
+                [password: yourPasswordHere]
+                [enable_encrypted_communication: True/False]
+                [certificates:
+                    broker_ca_cert: /path/to/pem/encoded/broker_ca_certificate.crt
+                    client_cert: /path/to/pem/encoded/client_certificate.crt
+                    client_key: /path/to/pem/encoded/client_certificate.key
+                    validate_broker_cert: True/False
+                    tls_version: ['tlsv1.2', 'tlsv1.1', 'tlsv1']
+                ]
+                [qos: [0,1,2] ]
+                [retain: True/False]
+    """
+
+    schema = {
+        'type': 'object',
+        'properties': {
+            'topic': {'type': 'string'},
+            'broker_address': {'type': 'string'},
+            'broker_port': {'type': 'integer', 'default': 1883},
+            'broker_timeout': {'type': 'integer', 'default': 30},
+            'broker_transport': {'type': 'string', 'default': 'tcp', 'enum': ['tcp', 'websockets']},
+            'broker_protocol': {'type': 'string', 'default': 'MQTTv311', 'enum': ['MQTTv31', 'MQTTv311']},
+            'username': {'type': 'string'},
+            'password': {'type': 'string'},
+            'enable_encrypted_communication': {'type': 'boolean', 'default': False},
+            'certificates': {
```

Comment:
should `certificates` have required properties? Reading it's usage below it would appear so

Reply:
The way I understand it, certificates are only required in certain cases
- A user wants to use public/private certs as an authentication mechanism (which requires CA, client_pub, and client_private)
- A user wants to enable SSL style encrypted communications with broker only, and not as a form of auth (which requires only CA cert)

But certs are not required if:
- using only user/pass for authentication
- or need no authentication to publish
- and/or aren't using encrypted communication channel.

Because of this, I just thought it best to let the user provide certificates as it fits their needs without trying to mandate anything.  Error on the side of flexibility over complex mandates

Reply:
Leaving as-is for flexibility around how certs may or may not be needed (depending on MQTT broker setup).